### PR TITLE
Better Helm App Operator

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

This basically rewrites the entire project. Adds source code for `helm-app-operator`. Initially generated using `operator-sdk new`, then added Helm logic from https://github.com/coreos/lostromos (`helmctrl` package).

### Testing

Follow the instructions in the `README.md`.

Addresses https://jira.coreos.com/browse/ALM-636

### CODE REVIEW

File: helm-app-operator/pkg/helm/helm.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+func chartDir() string {
+	return os.Getenv(HelmChartEnvVar)
+}
+
+func valuesFromResource(r *unstructured.Unstructured) ([]byte, error) {
+	re := map[string]interface{}{
+		"resource": map[string]interface{}{
+			"name":      r.GetName(),
+			"namespace": r.GetNamespace(),
+			"spec":      r.Object["spec"]}}
```
Comment (jzelinskie) [inline helm-app-operator/pkg/helm/helm.go]: move the final } to the next line

File: README.md [hunk 2]
```diff
@@ -2,8 +2,70 @@ This project is a component of the [Operator Framework](https://github.com/opera
 
 # Helm App Operator Kit
 
-This repository serves as a template for easily creating managed stateless applications that run either Kubernetes Deployments or Helm charts. It builds upon the [Lostromos project](https://github.com/wpengine/lostromos).
+This repository serves as a template for easily creating managed stateless applications that run either Kubernetes Deployments or Helm charts. It was inspired by the [Lostromos project](https://github.com/wpengine/lostromos). The underlying Operator was created using the `operator-sdk new` command.
```
Comment (ecordell) [inline README.md]: link to operator-sdk?

File: README.md [hunk 3]
```diff
@@ -2,8 +2,70 @@ This project is a component of the [Operator Framework](https://github.com/opera
...
+$ git checkout git@github.com:operator-framework/helm-app-operator-kit.git && cd helm-app-operator-kit
+$ docker build -t quay.io/<namespace>/<chart>-operator -e HELM_CHART=/path/to/helm/chart -e API_GROUP=<group/version> -e KIND=<Kind> .
+$ docker push quay.io/<namespace>/<chart>-operator
+```
+
+2) Modify the following Kubernetes YAML manifest files in `helm-app-operator/deploy`:
+
+File                          | Action
+------------------------------|--------------------------------------------------------------------------------------------------------
+`deploy/crd.yaml`             | Define your CRD (`kind`, `spec.version`, `spec.group` *must* match the `docker build` args)
```
Comment (ecordell) [inline README.md]: Why do we need `deploy/crd.yaml` and `deploy/olm-catalog/crd.yaml`?

Comment (alecmerdler) [reply README.md]: The `operator-sdk build` command generates both, I think it's so you can `kubectl create -f `deploy/olm-catalog` and ignore the upper level files.

**Comment by alecmerdler** [comment]
@ecordell @jzelinskie Updated and ready for review. Have confirmed it works and runs on a real cluster.

File: helm-app-operator/pkg/helm/helm.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	"github.com/operator-framework/helm-app-operator-kit/helm-app-operator/pkg/apis/app/v1alpha1"
+)
+
+const (
+	operatorName = "helm-app-operator"
+)
+
+// Controller can install and uninstall Helm releases given a custom resource
+// which provides runtime values for the Chart.
+type Controller interface {
```
Comment (ecordell) [inline helm-app-operator/pkg/helm/helm.go]: this is an odd name in the context of kubernetes where controller == has control loops.

File: helm-app-operator/version/version.go [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package version
+
+var (
+	Version = "0.0.1"
```
Comment (ecordell) [inline helm-app-operator/version/version.go]: Surely we're past 0.0.1 by now 😄

Comment (alecmerdler) [reply helm-app-operator/version/version.go]: Added by `operator-sdk new`

File: tomcat-operator/csv.yaml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
...Alx4mJ4YcAnw56MYMIQ7AKwzDTGnKEdrmFRulDXUejx1NSbFeTqG5oKPZnjbccFNHdZvi3PPE30j3X32L4CHSHOkc6QcbQ9X5GVpT97z/XBl0C+576m+jj2t1t8LwX/x9c6pB1iJW9pNSekycOddb/H1HatSXukENtsOr/UNOS/tabU+WLs13rdmSzy4e38y7Ci0rKDE73ULF5HydOO+JGu3xpk51aNHglp9U1tySCk2MObz4JvAXl0TD16/JGPh4tl+8Zvlg/aqjSM/v2pB4LltTUc9DaLAKtxatg53ulmOOw4qx/2KpJtvHiUKd1vschRv9g85yz/bl9yy+pMRPm5M5Pb0O/5whiQzQ6Jp4DMl9bUe6mo8TK82zc5D9ll7Wq1u4JOTBpZufbN07gzvPbcvDZkHum2efHlw+0BU3b29KXm89jCJO4fekfJ0Oe4+vhK4OmWEVcfxhsKdf21L2rzSech+e8PO+KG1n8Zz9ndakYBPyqIcDY8hsB3wewVTx5veti57ZkuHtf+JO7N3vP5BbOzACsqKcrTH7r42XFJRbLBq4wgr1sZeKC/W/9Q7eMJDVQG7cUPcwB3Qj7aV/cC7pFcw2ECHglUDUeeV7c3JptVbRrJ277PyvKbU8yMapi7I8Etqx5uBlg6r4TfLBw/y9Rxy3DD/ElgI7vz+oozLlswJCCHgrQ9jzqbPEk/3Djqb0zafG8pvAX9MXX/ATWAnc34OABtjcbX88zZr+5qt8dDOvclC0xBGfpZGQUSjptQIKcUF5UX6lX6vvLR/yKmxbPbjbqPjAldHQvKXs6Z6M7t7bZoOWKzbGlf7Oq2Xcb8+GYuolKdGr5Ooub4mI8D2eFK90tJufbJ2y0iwcW9ynGEIo7bcEOfWe435Z/gC88/05hfl6g2ftyXnDETVR0DH0ZSNhsIUKbnN0IWmlHvTspWwHV7AHfF8lyQAzA/65c0Ntea8c+p8vi+zuoA318d4a33sadxZ+jeMnd7/CXw3JQM4X9eYL4UwFKhUz60lkmor8DhjG0D+35T/AQq5KwDTXbxJAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDE4LTA3LTE0VDIxOjA1OjQ4LTA0OjAw7tyX+wAAACV0RVh0ZGF0ZTptb2RpZnkAMjAxOC0wNy0xNFQyMTowNTo0OC0wNDowMJ+BL0cAAAAASUVORK5CYII=
```
Comment (jzelinskie) [inline tomcat-operator/csv.yaml]: update this to svg

File: helm-app-operator/pkg/helm/engine.go [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/tiller/environment"
```
[the remark this answers is not in the document]
Comment (jzelinskie) [reply helm-app-operator/pkg/helm/engine.go]: apparently this is the URL we want to keep using

**Comment by ecordell** [comment]
@alecmerdler 

tomcat svg:

```
PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4NCjwhLS0gR2VuZXJhdG9yOiBBZG9iZSBJbGx1c3RyYXRvciAxNC4wLjAsIFNWRyBFeHBvcnQgUGx1Zy1JbiAuIFNWRyBWZXJzaW9uOiA2LjAwIEJ1aWxkIDQzMzYzKSAgLS0+DQo8IURPQ1RZUEUgc3ZnIFBVQkxJQyAiLS8vVzNDLy9EVEQgU1ZHIDEuMS8vRU4iICJodHRwOi8vd3d3LnczLm9yZy9HcmFwaGljcy9TVkcvMS4xL0RURC9zdmcxMS5kdGQiPg0KPHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IiB3aWR0aD0iMzAwcHgiDQoJIGhlaWdodD0iMjAwcHgiIHZpZXdCb3g9IjAgMCAzMDAgMjAwIiBlbmFibGUtYmFja2dyb3VuZD0ibmV3IDAgMCAzMDAgMjAwIiB4bWw6c3BhY2U9InByZXNlcnZlIj4NCjxnIGlkPSJDYWxxdWVfMyI+DQoJPGcgaWQ9IlhNTElEXzFfIj4NCgkJPHBhdGggZmlsbD0iI0QxQTQxQSIgZD0iTTUzLjcyOTk4MDUsMTMwYzIuOTU5OTYwOSwzLjY1MDM5MDYsNi4yNSw3LjM0OTYwOTQsOS44NzAxMTcyLDExLjA2OTMzNTkNCgkJCWMtMTAuNTQwMDM5MSwzLjE3MDg5ODQtMTcuNDUwMTk1Myw5LjA2MDU0NjktMjEuNzYwMjUzOSwxNC4yNjA3NDIyDQoJCQljLTYuODE5ODI0Miw4LjIzMDQ2ODgtMTAuNDI5Njg3NSwxOS4xMDA1ODU5LTkuNTk5NjA5NCwyOC40Mjk2ODc1SDE0LjA4OTg0MzgNCgkJCUMxNy43Mjk5ODA1LDE3NS44NjAzNTE2LDMyLjI1OTc2NTYsMTUzLjA0MDAzOTEsNTMuNzI5OTgwNSwxMzB6Ii8+DQoJCTxwYXRoIGZpbGw9IiNEMUE0MUEiIGQ9Ik0yNjMuOTkwMjM0NCwxODguNzA5OTYwOWgtMjIuMDcwMzEyNQ0KCQkJYy01MC42Mjk4ODI4LTI2LjQxOTkyMTktOTUuMDgwMDc4MS00NS4wNDk4MDQ3LTE0Mi4yOTk4MDQ3LTQ5LjEyMDExNzINCgkJCWMwLjEwMDA5NzctNi41NzAzMTI1LDEuMjcwMDE5NS0xNC4wMDk3NjU2LDMuNTIwMDE5NS0yMi4yOTk4MDQ3bC00LjM1MDA5NzctMS4xNjk5MjE5DQoJCQljLTIuMzEwMDU4Niw4LjUxOTUzMTMtMy41MjAwMTk1LDE2LjIyOTQ5MjItMy42NjAxNTYzLDIzLjEyOTg4MjhjLTQuOTE5OTIxOS0wLjMzMDA3ODEtOS44NzAxMTcyLTAuNDkwMjM0NC0xNC44NzAxMTcyLTAuNDkwMjM0NA0KCQkJYy00LjIyOTk4MDUsMC04LjA4OTg0MzgsMC4zOTA2MjUtMTEuNjA5ODYzMywxLjA3MDMxMjVjLTQuNTg5ODQzOC00LjU4MDA3ODEtOC41LTguOTU5OTYwOS0xMS44MTk4MjQyLTEzLjEwOTM3NQ0KCQkJYzIxLjc3OTc4NTItMjIuNjgwNjY0MSw1Ny42NDk5MDIzLTQzLjIwMTE3MTksODguODcwMTE3Mi01MC45MjA4OTg0DQoJCQljOC43OTAwMzkxLDI5LjIxOTcyNjYsMjcuNjYwMTU2Myw0OS4yNDAyMzQ0LDUwLjU1OTU3MDMsNjIuNzA5OTYwOWMxLjY3OTY4NzUtMC45NTk5NjA5LDMuMzMwMDc4MS0xLjk1OTk2MDksNC45NTAxOTUzLTMuMDE5NTMxMw0KCQkJTDI0MS4xNjAxNTYzLDE3OC41bDEuNzUtMC4xNTAzOTA2YzEwLjkwOTE3OTctMC45Mzk0NTMxLDE3LjEwOTM3NSw0Ljk4MDQ2ODgsMTguNzUsNi44MjAzMTI1DQoJCQlDMjYyLjg3MDExNzIsMTg2LjUwOTc2NTYsMjYzLjU4OTg0MzgsMTg3Ljc0MDIzNDQsMjYzLjk5MDIzNDQsMTg4LjcwOTk2MDl6Ii8+DQoJCTxwYXRoIGZpbGw9IiNGRkRDNzYiIGQ9Ik0yNDQuNSw0NC44NDk2MDk0YzAuODEwNTQ2OSwzOS42OTA0Mjk3LTE5LjQzOTQ1MzEsNzEuMjgwMjczNC00Ny43OTAwMzkxLDg3LjQ4MDQ2ODgNCgkJCWMtNDguMTIwMTE3Mi0yMy45Njk3MjY2LTU3LjE4MDE3NTgtODQuNzA5OTYwOS00OC45Nzk5ODA1LTExNy4yOTAwMzkxDQoJCQlDMTUwLjAwOTc2NTYsMjcuNSwxNTQuMTYwMTU2MywzNS4zMTA1NDY5LDE2My4xMjAxMTcyLDM5LjE0MDYyNWMxOS41NDAwMzkxLTUuNTkwODIwMyw0Ni40Mjk2ODc1LTYuMDMxMjUsNjYuMjQwMjM0NC0wLjU3MDMxMjUNCgkJCWM4LjI1LTYsMTEuNDUwMTk1My0xNS4xNjk5MjE5LDEzLjIwMDE5NTMtMjMuMTc5Njg3NUMyNDQuNzQwMjM0NCwyNC43MDk5NjA5LDI0NC40OTAyMzQ0LDQ0Ljg0OTYwOTQsMjQ0LjUsNDQuODQ5NjA5NHoiLz4NCgk8L2c+DQo8L2c+DQo8ZyBpZD0iTGF5ZXJfMSI+DQoJPHBvbHlnb24gcG9pbnRzPSIxMjMuNDU4NDk2MSw4Mi40Mjc3MzQ0IDEzMC44MDA3ODEzLDEwNC4xODM1OTM4IDEwOC41MDE0NjQ4LDg5LjIyNjU2MjUgCSIvPg0KCTxwb2x5Z29uIHBvaW50cz0iMTAxLjk3NTA5NzcsOTEuOTQ2Mjg5MSAxMDkuODYxMzI4MSwxMDUuNTQyOTY4OCA5My4yNzI5NDkyLDk3LjExMzI4MTMgCSIvPg0KCTxwb2x5Z29uIHBvaW50cz0iODQuMDI2ODU1NSwxMDIuNTUxNzU3OCA4Ny44MzM5ODQ0LDExNi4xNDk0MTQxIDc2LjQxMjU5NzcsMTA4LjI2MjY5NTMgCSIvPg0KCTxwb2x5Z29uIHBvaW50cz0iMjA1LjU4NDk2MDksMTM4Ljk5MjE4NzUgMTk4LjUxNDY0ODQsMTQ5Ljg3MDExNzIgMjExLjAyMzQzNzUsMTQ0LjE1OTE3OTcgCSIvPg0KCTxwb2x5Z29uIHBvaW50cz0iMjE2LjczNDM3NSwxNTEuNTAxOTUzMSAyMDUuMzEzNDc2NiwxNjAuMjA0MTAxNiAyMjEuMDg1OTM3NSwxNTYuMTI1IAkiLz4NCgk8cG9seWdvbiBwb2ludHM9IjE4OC45OTYwOTM4LDMzLjQ3ODUxNTYgMTk3LjQyNjc1NzgsNTEuOTcwNzAzMSAyMDIuODY1MjM0NCwzMy4yMDcwMzEzIAkiLz4NCgk8cG9seWdvbiBwb2ludHM9IjE0Ni4zMDE3NTc4LDYwLjQwMDM5MDYgMTU5LjYyNjk1MzEsNjcuNDcwNzAzMSAxNDcuNjYxMTMyOCw2OC4yODcxMDk0IAkiLz4NCgk8cG9seWdvbiBwb2ludHM9IjE0OS41NjQ5NDE0LDc0LjgxNDQ1MzEgMTU3LjcyMzE0NDUsNzguMDc3MTQ4NCAxNTAuOTI0ODA0Nyw3OS45ODA0Njg4IAkiLz4NCgk8cG9seWdvbiBwb2ludHM9IjI0NC43NDUxMTcyLDY1LjU2ODM1OTQgMjMzLjg2NzE4NzUsNzAuMTkxNDA2MyAyNDIuODQwODIwMyw3MS41NTA3ODEzIAkiLz4NCgk8cG9seWdvbiBwb2ludHM9IjI0MC4zOTM1NTQ3LDc4Ljg5MjU3ODEgMjMxLjk2Mzg2NzIsODEuMzQwODIwMyAyMzguNzYxNzE4OCw4My43ODgwODU5IAkiLz4NCjwvZz4NCjxnIGlkPSJDYWxxdWVfNCI+DQoJPHJlY3QgeD0iMTQxLjkyMzMzOTgiIHk9Ijk5LjMwMjczNDQiIHdpZHRoPSIyNi4yNDQ2Mjg5IiBoZWlnaHQ9IjMuMjM5MjU3OCIvPg0KCTxyZWN0IHg9IjE0MS45MjMzMzk4IiB5PSI5MS41MjUzOTA2IiB3aWR0aD0iMjYuMjQ0NjI4OSIgaGVpZ2h0PSIzLjI0MDIzNDQiLz4NCgk8cmVjdCB4PSIyMjIuMDYzNDc2NiIgeT0iOTkuMzAyNzM0NCIgd2lkdGg9IjI2LjI0NDE0MDYiIGhlaWdodD0iMy4yMzkyNTc4Ii8+DQoJPHJlY3QgeD0iMjIyLjA2MzQ3NjYiIHk9IjkxLjUyNTM5MDYiIHdpZHRoPSIyNi4yNDQxNDA2IiBoZWlnaHQ9IjMuMjQwMjM0NCIvPg0KCTxnPg0KCQk8cGF0aCBkPSJNMjEyLjgxMzQ3NjYsNzIuMzE2NDA2M2gtMjcuMTg3NWgtMy4wMjA1MDc4aC0xMi41NjA1NDY5di0yLjkxNjAxNTZoMTMuMzA5NTcwMw0KCQkJYzEuNjM4NjcxOS03LjAxNzU3ODEsMS45NjI4OTA2LTEzLjcyNDYwOTQtMC4wMTM2NzE5LTE3LjEyNWMtMC44NDM3NS0xLjQ0NTMxMjUtMi4wMDk3NjU2LTIuMTIxMDkzOC0zLjY3MzgyODEtMi4xMjEwOTM4DQoJCQljLTcuNjA5Mzc1LDAtMTAuNzUyOTI5Nyw4LjA0Njg3NS0xMC44ODM3ODkxLDguMzg4NjcxOWwwLjAwMTk1MzEtMC4wMDE5NTMxbC0yLjczMDQ2ODgtMS4wMjQ0MTQxDQoJCQljMC4xNTYyNS0wLjQxOTkyMTksMy45NjQ4NDM4LTEwLjI3ODMyMDMsMTMuNjEyMzA0Ny0xMC4yNzgzMjAzYzIuNjkyMzgyOCwwLDQuODMzOTg0NCwxLjIzNjMyODEsNi4xOTE0MDYzLDMuNTcwMzEyNQ0KCQkJYzIuNDEwMTU2Myw0LjE0MDYyNSwyLjEyNjk1MzEsMTEuMzA0Njg3NSwwLjQ5NDE0MDYsMTguNTkxNzk2OWgyMy4zNTQ0OTIyDQoJCQljMy4xMDI1MzkxLTkuMTE1MjM0NCw5LjU4MTA1NDctMTMuNDEzMDg1OSwyMC40MDUyNzM0LTEzLjQxMzA4NTl2Mi45MTUwMzkxDQoJCQljLTExLjczMjQyMTksMC0xNS4wMTg1NTQ3LDQuOTczNjMyOC0xNy4zNjYyMTA5LDEwLjQ5ODA0NjloMTIuNzQzMTY0MWwtMC4wMjkyOTY5LDIuOTAxMzY3MkwyMTIuODEzNDc2Niw3Mi4zMTY0MDYzeiIvPg0KCQk8cGF0aCBkPSJNMjE0LjczMDQ2ODgsODYuNzE3NzczNGwwLjAxODU1NDctMC4wMDI5Mjk3Yy0zLjQ1ODk4NDQtNS4xMDA1ODU5LTQuNTU0Njg3NS05LjQ1NTA3ODEtMy4xMDgzOTg0LTE0LjQxMzA4NTkNCgkJCWwtMi45NzA3MDMxLDAuMDE0NjQ4NGMtMS4wMzUxNTYzLDMuMzAwNzgxMy0wLjYyMDExNzIsOC4yNzM0Mzc1LDEuOTI4NzEwOSwxMi41NDAwMzkxaC0yOS42NzQ4MDQ3DQoJCQljMS44MDU2NjQxLTMuNjE2MjEwOSwzLjQ3OTQ5MjItOC4wMjUzOTA2LDQuNzAyMTQ4NC0xMi41NDAwMzkxaC0zLjAyMDUwNzgNCgkJCWMtMS4zNDc2NTYzLDQuNzg3MTA5NC0zLjI0MTIxMDksOS41MjQ0MTQxLTUuMzcyMDcwMywxMy4yNzczNDM4bC0wLjY4OTQ1MzEsMS4yMTI4OTA2bDE2LjY1MjM0MzgsMTAuNDgyNDIxOWwtOS4zNzUsNi4xNzc3MzQ0DQoJCQlsMS42MDU0Njg4LDIuNDM1NTQ2OWwxMC40Nzg1MTU2LTYuOTA4MjAzMWwxMS4zMTE1MjM0LDcuMzgxODM1OWwxLjU1MzcxMDktMi40Njc3NzM0bC0xMC40ODgyODEzLTYuNDg4MjgxMw0KCQkJYzAsMCwxNS42ODE2NDA2LTEwLjE4NjUyMzQsMTYuNDYwOTM3NS0xMC42ODM1OTM4QzIxNC43MzgyODEzLDg2LjcyOTQ5MjIsMjE0LjczNTM1MTYsODYuNzIzNjMyOCwyMTQuNzMwNDY4OCw4Ni43MTc3NzM0eg0KCQkJIE0xOTUuODcwMTE3Miw5NS41MjYzNjcybC0xMi4zMTczODI4LTcuNzU0ODgyOGwyNC4wNzEyODkxLDAuMDA1ODU5NEwxOTUuODcwMTE3Miw5NS41MjYzNjcyeiIvPg0KCTwvZz4NCjwvZz4NCjxnIGlkPSJDYWxxdWVfMiI+DQoJPHBhdGggZD0iTTI0OS4zOTQwNDMsNDEuMzMzMDA3OGMtMC4wMTg1NTQ3LTUuMDcwMzEyNS0xLjU5NzY1NjMtMjYuNS00LjUtMzIuNDAwMzkwNg0KCQljLTcuODY5MTQwNiwyLjk5MDIzNDQtMjIuMTg5NDUzMSwxMi42Nzk2ODc1LTI1LjQzMDY2NDEsMjIuNTg3ODkwNg0KCQljLTE1LjUzMjIyNjYtMi4zMTkzMzU5LTMzLjA0MTk5MjItMi4wNDQ5MjE5LTQ4LjM4MDg1OTQsMC42MDkzNzVjLTIuNTYyNS0xMC42NjAxNTYzLTE1LjQyNDMxNjQtMTguODE4MzU5NC0yNS4xMjc0NDE0LTIzLjQxNDA2MjUNCgkJYy00LjA5Mjc3MzQsNi45MDYyNS01LjkwNjI1LDIyLjkxOTkyMTktNS40NDQ4MjQyLDM0LjI3OTI5NjljLTAuMDE3NTc4MSwwLjAxMTcxODgtMC4wMzY2MjExLDAuMDIzNDM3NS0wLjA1NDE5OTIsMC4wMzYxMzI4DQoJCWMxLjc5OTgwNDcsNDcuNzAwMTk1MywyNS4xOTk3MDcsNzcuMzk5NDE0MSw1NS43OTkzMTY0LDk1LjM5OTQxNDENCgkJYzMxLjQ4MTQ0NTMtMTcuOTg4MjgxMyw1My45NzE2Nzk3LTUzLjA1NjY0MDYsNTMuMTAwNTg1OS05Ny4xMTkxNDA2DQoJCUMyNDkuMzY4NjUyMyw0MS4zMTkzMzU5LDI0OS4zODIzMjQyLDQxLjMyNTE5NTMsMjQ5LjM5NDA0Myw0MS4zMzMwMDc4eiBNMjIzLjE5Nzc1MzksMzIuMTM2NzE4OA0KCQljMC4xMTQyNTc4LDAuMDIwNTA3OCwwLjIyODUxNTYsMC4wNDE5OTIyLDAuMzQzNzUsMC4wNjM0NzY2DQoJCUMyMjMuNDI3MjQ2MSwzMi4xNzg3MTA5LDIyMy4zMTI5ODgzLDMyLjE1NzIyNjYsMjIzLjE5Nzc1MzksMzIuMTM2NzE4OHogTTIyMS4yMzk3NDYxLDMxLjc5ODgyODENCgkJYzAuMjc1MzkwNiwwLjA0Mzk0NTMsMC41NTM3MTA5LDAuMDkzNzUsMC44MzAwNzgxLDAuMTQwNjI1QzIyMS43OTM0NTcsMzEuODkyNTc4MSwyMjEuNTE3MDg5OCwzMS44NDQ3MjY2LDIyMS4yMzk3NDYxLDMxLjc5ODgyODF6DQoJCSBNMjQ0LjQ5NjU4Miw0NC43NzQ0MTQxYzAuODA5NTcwMywzOS42OTA0Mjk3LTE5LjQ0MDQyOTcsNzEuMjc5Mjk2OS00Ny43OTAwMzkxLDg3LjQ4MDQ2ODgNCgkJYy00OC4xMTgxNjQxLTIzLjk3NjU2MjUtNTcuMTgyNjE3Mi04NC43MTA5Mzc1LTQ4Ljk3NzA1MDgtMTE3LjI4OTA2MjUNCgkJYzIuMjgyNzE0OCwxMi40NTUwNzgxLDYuNDI0MzE2NCwyMC4yNjY2MDE2LDE1LjM5MzU1NDcsMjQuMDk4NjMyOGMxOS41MzI3MTQ4LTUuNTkwODIwMyw0Ni40MjMzMzk4LTYuMDMyMjI2Niw2Ni4yMzA5NTctMC41NjkzMzU5DQoJCWM4LjI1NDg4MjgtNi4wMDE5NTMxLDExLjQ1NjA1NDctMTUuMTY5OTIxOSwxMy4yMDQxMDE2LTIzLjE3OTY4NzUNCgkJQzI0NC43NDE2OTkyLDI0LjYzNjcxODgsMjQ0LjQ4OTc0NjEsNDQuNzcwNTA3OCwyNDQuNDk2NTgyLDQ0Ljc3NDQxNDF6Ii8+DQoJPHBhdGggZD0iTTI2NS4wMTcwODk4LDE4Mi4wOTA4MjAzYy00Ljk0MTQwNjMtNS41MjgzMjAzLTE0LjYyMzA0NjktOC4xNDQ1MzEzLTIyLjcwNzAzMTMtNy45NTExNzE5DQoJCWMtNS43MzgyODEzLTYuODMwMDc4MS0zOS4wMDM5MDYzLTQxLjg2MTMyODEtMzkuMDAzOTA2My00MS44NjEzMjgxbC0yLjMwNTY2NDEsMi45MDMzMjAzbDQwLjE2MjEwOTQsNDMuMjM4MjgxMw0KCQlsMS43NDMxNjQxLTAuMTQ5NDE0MWMxMC45MTIxMDk0LTAuOTM0NTcwMywxNy4xMTUyMzQ0LDQuOTgzMzk4NCwxOC43NTY4MzU5LDYuODE5MzM1OQ0KCQljMS4yMDQxMDE2LDEuMzQ2Njc5NywxLjkzMDY2NDEsMi41NzUxOTUzLDIuMzI2MTcxOSwzLjUzOTA2MjVoLTIyLjA3NTE5NTMNCgkJYy01MC42MjQwMjM0LTI2LjQxNjAxNTYtOTUuMDc4MTI1LTQ1LjA0Mzk0NTMtMTQyLjI5NzM2MzMtNDkuMTEyMzA0Nw0KCQljMC4xMDM1MTU2LTYuNTcxMjg5MSwxLjI3MzQzNzUtMTQuMDA5NzY1NiwzLjUxODA2NjQtMjIuMjk4ODI4MWwtNC4zNDMyNjE3LTEuMTc2NzU3OA0KCQljLTIuMzA3NjE3Miw4LjUyMDUwNzgtMy41MjM0Mzc1LDE2LjIzNjMyODEtMy42NjA2NDQ1LDIzLjEzMjgxMjVjLTQuOTE5OTIxOS0wLjMyNjE3MTktOS44NzIwNzAzLTAuNDk1MTE3Mi0xNC44Njc2NzU4LTAuNDk1MTE3Mg0KCQljLTQuMjM2ODE2NCwwLTguMDk1MjE0OCwwLjM5MzU1NDctMTEuNjE0MjU3OCwxLjA3NzE0ODRjLTQuNTg5ODQzOC00LjU4NjkxNDEtOC41LTguOTU4OTg0NC0xMS44MjI3NTM5LTEzLjEwODM5ODQNCgkJYzIxLjkzNjAzNTItMjIuODQ5NjA5NCw1OC4xNDk5MDIzLTQzLjQ5ODA0NjksODkuNTMzMjAzMS01MS4wOTE3OTY5bC0xLjAxNTEzNjctNC4zOTY0ODQ0DQoJCWMtMzMuNzkxOTkyMiw4LjE0ODQzNzUtNzAuNzc5Nzg1MiwzMC4zMzQ5NjA5LTkxLjM5MDYyNSw1MS43NjE3MTg4DQoJCWMtOS41NjQ5NDE0LTEyLjkxMDE1NjMtMTMuMzU5ODYzMy0yMy41MDM5MDYzLTE0LjQ4NjgxNjQtMzEuNTMyMjI2Ng0KCQljLTEuNDIzODI4MS0xMC4xMzk2NDg0LDAuOTk2NTgyLTE5LjQ0MTQwNjMsNi45OTkwMjM0LTI2Ljg5OTQxNDFjOS4zOTY5NzI3LTExLjY3MzgyODEsMjIuNzQ5MDIzNC0xMi40OTYwOTM4LDM2LjQ3NzA1MDgtMTAuMTg3NQ0KCQljLTAuMjY4MDY2NCwyLjE2Nzk2ODgsMC4wMTcwODk4LDQuMjQwMjM0NCwwLjg4NTI1MzksNS41MjI0NjA5YzMuNjMwODU5NCw1LjM2MjMwNDcsMjMuMTQzNTU0Nyw3LjI0NjA5MzgsMzQuNzkxMDE1NiwyLjA0OTgwNDcNCgkJYy04LjU5NTIxNDgtMTIuMDQ0OTIxOS0yNi4wMDU4NTk0LTE3LjkyNTc4MTMtMzAuODMwMDc4MS0xNS41NjkzMzU5DQoJCWMtMS41OTgxNDQ1LDAuNzgwMjczNC0yLjgwMzcxMDksMi4yMTM4NjcyLTMuNjI5ODgyOCwzLjg4NTc0MjJjLTQuNzQ1MTE3Mi0wLjg0ODYzMjgtOS40NTgwMDc4LTEuNDgwNDY4OC0xMi45MDI4MzItMS40NTUwNzgxDQoJCWMtMTIuMTA3NDIxOSwwLjA4ODg2NzItMjEuMzYzMjgxMyw0LjMxOTMzNTktMjguMjk1ODk4NCwxMi45MzI2MTcyDQoJCWMtNi44MDE3NTc4LDguNDUwMTk1My05LjU1MDc4MTMsMTguOTQ0MzM1OS03Ljk1MDE5NTMsMzAuMzQ2Njc5Nw0KCQljMS41MTYxMTMzLDEwLjc5OTgwNDcsNi44NjMyODEzLDIyLjI4NjEzMjgsMTUuODI1MTk1MywzNC4xOTgyNDIyYy0xLjQ4ODc2OTUsMS41OTY2Nzk3LTIuOTI3NzM0NCwzLjE3MzgyODEtNC4zMTI1LDQuNzIyNjU2Mw0KCQljLTIxLjg0MzI2MTcsMjQuNDQxNDA2My0zNy44MTAwNTg2LDUwLjY1MDM5MDYtMzcuODEwMDU4Niw1NC45OTMxNjQxdjIuMjVoMjguODE1OTE4bC0wLjUwNDg4MjgtMi42Njc5Njg4DQoJCWMtMS42Mjc0NDE0LTguNjAwNTg1OSwxLjYyMjU1ODYtMTkuMzUwNTg1OSw4LjI3OTI5NjktMjcuMzg0NzY1Ng0KCQljNC4xNzkxOTkyLTUuMDQzOTQ1MywxMS4wOTk2MDk0LTEwLjgwMDc4MTMsMjEuOTY3NzczNC0xMy40NTcwMzEzDQoJCWM5LjY2MzA4NTksOS40MzA2NjQxLDIxLjk3NDYwOTQsMTkuNDQ2Mjg5MSwzNS42Nzg3MTA5LDI5LjEwOTM3NWgyMS40MTM1NzQydi0yLjI1DQoJCWMwLTIuNTM5MDYyNS0xLjI1ODc4OTEtNC43NjE3MTg4LTMuNTQ0OTIxOS02LjI1NjgzNTljLTMuMzgyODEyNS0yLjIxMTkxNDEtOC42MDQ5ODA1LTIuNTE2NjAxNi0xMy40MDE4NTU1LTAuODgwODU5NA0KCQljLTQuNTczMjQyMi01LjA5Mjc3MzQtNy4xMzcyMDctMTEuODk2NDg0NC03LjY5MDkxOC0yMC4zNDY2Nzk3DQoJCWM0Ni40NzQxMjExLDQuMTE1MjM0NCw5MC45MjY3NTc4LDIyLjg4MjgxMjUsMTQxLjA4NDk2MDksNDkuMDg0OTYwOWgyNy41OTc2NTYzbDAuMzYxMzI4MS0xLjgwODU5MzgNCgkJQzI2OS4yODQ2NjgsMTg4Ljc0NTExNzIsMjY3Ljg4MTM0NzcsMTg1LjI5NDkyMTksMjY1LjAxNzA4OTgsMTgyLjA5MDgyMDN6IE00MS44Mzc4OTA2LDE1NS4yNTQ4ODI4DQoJCWMtNi44MTU5MTgsOC4yMjY1NjI1LTEwLjQyODcxMDksMTkuMDkzNzUtOS42MDEwNzQyLDI4LjQyMzgyODFoLTE4LjE1MjgzMg0KCQljMy42NDc0NjA5LTcuODkzNTU0NywxOC4xNzYyNjk1LTMwLjcxMTkxNDEsMzkuNjQzNTU0Ny01My43NTk3NjU2YzIuOTU4MDA3OCwzLjY1NjI1LDYuMjUyOTI5Nyw3LjM0OTYwOTQsOS44NzIwNzAzLDExLjA3MzI0MjINCgkJQzUzLjA1OTU3MDMsMTQ0LjE2MDE1NjMsNDYuMTQ5NDE0MSwxNTAuMDUxNzU3OCw0MS44Mzc4OTA2LDE1NS4yNTQ4ODI4eiIvPg0KPC9nPg0KPC9zdmc+DQo=
```

**Pull request merged**